clarity/contracts/arkadiko-dao.clar

```diff
@@ -112,6 +112,22 @@
   )
 )
 
+;; adds a new contract, only new ones allowed
+(define-public (add-contract-address (name (string-ascii 256)) (address principal) (qualified-name principal))
+  (begin
+    (asserts! (is-eq tx-sender (var-get dao-owner)) (err ERR-NOT-AUTHORIZED))
```

Comment:
REQ: By putting this method in the DAO, the protocol code become centralised. We could undo governance changes and deploy malicious contracts at any time. The DAO can not change and we need to be the 'dao-owner' to perform external tasks, so platform can not become fully decentralised?

My suggestion would be to put this feature in the governance contract. The 'dao-owner' can call the method on governance, and governance can call the existing 'set-contract-address' on the DAO.

Reply:
@nieldeckx yes, it's true that this is a bit more centralised and the people who own the `dao-owner` address will be able to add new contracts to the protocol.

However, they cannot just overwrite other contract keys & values in the map. Look at the difference between `map-set`, which allows overwriting keys, vs `map-insert`, which returns false and does not set the new key/value pair when it already exists. So this method is addition only, no?

Reply:
Ok, it's true the method can only be used to add new contracts but still provides a superpower which can not be removed later.

Reply:
@nieldeckx ya - agree. 

I can move it to the governance contract, and keep the functionality the same, with the purpose that the governance contract can be replaced (and thus the functionality removed)?

Reply:
@nieldeckx updated with your feedback! what do you think?

Reply:
@philipdesmedt perfect!